vyper/semantics/types/function.py

```diff
@@ -93,10 +131,34 @@ def __init__(
             "default_return_value": KwargSettings(return_type, None),
         }
 
+        self.gas_estimate = None
+
+        # we could do a bit better than this but it just needs to be unique
+        visibility = "internal" if self.is_internal else "external"
+        argz = ",".join([str(argtyp) for argtyp in self.arguments_typs])
+        ir_identifier = mkalphanum(f"{visibility} {self.name} ({argz})")
+        self.ir_info: FunctionIRInfo = (
+            InternalFunctionIRInfo(ir_identifier)
+            if self.is_internal
+            else ExternalFunctionIRInfo(ir_identifier)
```

Comment:
ir_info could maybe be set much later - during codegen/module.py for instance.

Reply:
done in https://github.com/vyperlang/vyper/pull/3390/commits/08a72458c8ec308b0e8d886c64bd824da4dbecc9, but requires quite a few mypy hints to be added